qml/pages/FirstPage.qml

```diff
@@ -5,26 +5,47 @@ import "../js/yleApi.js" as YleApi
 
 Page {
     id: page
+    property real rowCount
+    property real columnCount: isPortrait ? 1.0 : 2.0
+
+    property int currIndex: -1
+    property int firstOffsetIndex
+    property real offsetY
+
+    property int imageSizeX
+    property int imageSizeY
 
     Component.onCompleted: {
+        // This prevents image reload on orientation change
+        imageSizeX = Math.min(width, height)
+        imageSizeY = Math.floor(imageSizeX / 16.0 * 9.0)
+
         updateCover(qsTr("Current Broadcasts"), "", "")
         YleApi.getCurrentBroadcasts()
             .then(function(broadcasts) {
-                listView.model = broadcasts
+                gridView.model = broadcasts
             })
     }
 
     onVisibleChanged: {
-        if (visible) updateCover(qsTr("Current Broadcasts"), "", "")
+        if (visible)
+            updateCover(qsTr("Current Broadcasts"), "", "")
     }
 
     // The effective value will be restricted by ApplicationWindow.allowedOrientations
     allowedOrientations: Orientation.All
 
-    // To enable PullDownMenu, place our content in a SilicaFlickable
-    SilicaFlickable {
+    onOrientationChanged: rowCount = Math.ceil(model.length / columnCount)
```

Comment:
model is a variable of gridView so should be gridView.model here

Reply:
Correct, I'll fix that.